src/types/json.h

```diff
@@ -65,6 +66,47 @@ struct JsonValue {
     return Status::OK();
   }
 
+  Status StrAppend(std::string_view path, const std::string &append_value, std::vector<uint64_t> &append_cnt) {
+    try {
+      auto append_trimmed = util::Trim(append_value, "\"");
+      jsoncons::jsonpath::json_replace(
+          value, path, [&append_trimmed, &append_cnt](const std::string & /*path*/, jsoncons::json &origin) {
+            if (origin.is_string()) {
+              auto origin_str = origin.to_string();
+              // need trim "
+              auto origin_trimmed = util::Trim(origin_str, "\"");
+              append_cnt.push_back(origin_trimmed.length() + append_trimmed.length());
+              std::string new_value;
+              new_value.append("\"").append(origin_trimmed).append(append_trimmed).append("\"");
+              origin = jsoncons::json::parse(new_value);
+            } else {
+              append_cnt.push_back(0);
+            }
+          });
+    } catch (const jsoncons::jsonpath::jsonpath_error &e) {
+      return {Status::NotOK, e.what()};
+    }
+
+    return Status::OK();
+  }
+
+  Status StrLen(std::string_view path, std::vector<uint64_t> &str_lens) const {
+    try {
+      jsoncons::jsonpath::json_query(value, path,
+                                     [&str_lens](const std::string & /*path*/, const jsoncons::json &origin) {
+                                       if (origin.is_string()) {
+                                         // need subtraction "
+                                         str_lens.push_back(origin.to_string().length() - 2);
+                                       } else {
+                                         str_lens.push_back(0);
```

Comment:
How can you distinguish between zero-sized array and non-array here?

Reply:
no need distinguish? I don't quite understand what you mean.
`127.0.0.1:6666> json.set doc $ '{"a":"foo\n", "nested": {"a": "hello"}, "nested2": {"a": 31}, "nested3": [{"a": "world"}], "nested4": {"a": []}}'
OK
127.0.0.1:6666> json.strlen doc $..a
1) (integer) 4
2) (integer) 5
3) (nil)
4) (integer) 5
5) (nil)
`

Reply:
```
> json.set x $ '["", 1]'
OK
> json.strlen x $[0]
1) (integer) 0
> json.strlen x $[1]
1) (nil)
```

Reply:
I understand, need distinguish between null and zero, zero is not null